mirage/server/dns_server_mirage.ml

```diff
@@ -187,12 +187,15 @@ module Make (P : Mirage_clock.PCLOCK) (M : Mirage_clock.MCLOCK) (TIME : Mirage_t
       let elapsed = M.elapsed_ns () in
       let state', out = Dns_server.Secondary.closed !state now elapsed ip in
       state := state' ;
-      request (ip, out)
-    and read_and_handle ip f =
+      if not timer then
+        request ~timer (ip, out)
+      else
+        Lwt.return_unit
```

Comment:
At an earlier iteration this was clearer, but `close` is only called with `~timer:true` through `request ~timer:true` (again through `timer`). We could move this code into `request`:

```OCaml
close _ >>= fun () ->
if timer then Lwt.return_unit else request _
```

I had written this comment once before but it seems it was not submitted(?!)

Reply:
Hmm?

`close` is called from various points in the code, and the `~timer` can be true or false (indicating whether the connection establishment originated from the timer). Now, when `timer` is false, then `close` calls `request` again. (I'm still not sure whether this is wise -- but there are some (i.e. let's-encrypt-secondary) unikernels that rely on the fact that the TCP connection is always established (to receive NOTIFY events)...

Reply:
Nevermind, this is not the case anymore since I believe 8e152f66b440de7716d148d59cebb626cfadea85.